src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -253,6 +256,12 @@ public async Task ProcessInputAsync(Message message, Flow flow, IContext message
                                     // Store the previous state
                                     await _stateManager.SetPreviousStateIdAsync(context, previousStateId, linkedCts.Token);
 
+                                    if (previousStateId.StartsWith(Constants.DESK_STATE_PREFIX) && previousStateId != state.Id)
```

Comment:
Suggestion: how to check if a state is the "Desk state" should be encapsulated. This could be a method (named for instance `InDesk`) inside State class itself or maybe an extension method if you like.